lib/util/BuilderLifecycle.js

```diff
@@ -0,0 +1,23 @@
+const Shell = require('shelljs');
+const UserInteractor = require('./UserInteractor');
+
+function executeCommands(commands, commandVariables = {}) {
+    if (!commands || commands.length == 0) {
+        return;
+    }
+
+    for (var command of commands) {
+        const finalCommand = Object.keys(commandVariables).reduce((str, variable) => {
+            return str.replace(new RegExp(`\\\${${variable}}`, RegExp.global), commandVariables[variable]);
+        }, command);
+        if (Shell.exec(finalCommand).code !== 0) {
+            UserInteractor.processError({message: "Post-build command failure."})
```

Comment:
I think this is used for both "preBuild" and "postDeviceBuild", isn't this going to print "Post-build command failure." for both cases?

Reply:
Great catch! I pulled this common stuff out at the end and totally whiffed on the error message. I think the `executeCommands` function to take a `label` parameter or something.